File: scripts/python/parsers/static_resource_parser.py
"""Parser for Static Resource metadata (.resource-meta.xml)."""
import re
from pathlib import Path


def _extract_tag(xml: str, tag: str) -> str | None:
    m = re.search(rf'<{tag}[^>]*>([\s\S]*?)</{tag}>', xml)
    return m.group(1).strip() if m else None


def parse_static_resources(type_dir: str) -> dict:
    """
    @param type_dir: path to the staticresources/ folder
    @returns: { resources: list }
    """
    resources = []
    p = Path(type_dir)
    if not p.exists():
        return {'resources': resources}

    for f in sorted(p.glob('*.resource-meta.xml')):
        xml = f.read_text(encoding='utf-8')
        resources.append({
            'apiName':      f.name.replace('.resource-meta.xml', ''),
            'contentType':  _extract_tag(xml, 'contentType'),
            'cacheControl': _extract_tag(xml, 'cacheControl'),
        })

    return {'resources': resources}

File: scripts/node/parsers/staticResourceParser.js
'use strict';

const fs   = require('fs');
const path = require('path');

function extractTag(xml, tag) {
  const m = xml.match(new RegExp(`<${tag}[^>]*>([\\s\\S]*?)<\\/${tag}>`));
  return m ? m[1].trim() : null;
}

/**
 * Reads Static Resource metadata from SFDX source format.
 * @param {string} typeDir - path to the staticresources/ folder
 * @returns {{ resources: Array }}
 */
async function parseStaticResources(typeDir) {
  const resources = [];
  if (!fs.existsSync(typeDir)) return { resources };

  const files = fs.readdirSync(typeDir).filter(f => f.endsWith('.resource-meta.xml'));

  for (const file of files) {
    const xml = fs.readFileSync(path.join(typeDir, file), 'utf8');
    resources.push({
      apiName:      file.replace('.resource-meta.xml', ''),
      contentType:  extractTag(xml, 'contentType'),
      cacheControl: extractTag(xml, 'cacheControl'),
    });
  }

  return { resources };
}

module.exports = { parseStaticResources };
